### PR TITLE
Add start timestamp

### PR DESCRIPTION
**Pull request by colmsnowplow**
- Pulls in the most recent commit from upstream (which improves error handling)
- Adds a 'start from timestamp' configuration option

Manually tested by pulling my local branch into the stream-replicator project. Ideally we'd set up a better test rig in this repo, but I feel that's better done as part of the further planned work on this library, since it could get quite involved.

### CODE REVIEW

File: shard_consumer.go [hunk 1]
```diff
@@ -46,6 +50,8 @@ func getShardIterator(k kinesisiface.KinesisAPI, streamName string, shardID stri
 		ShardIteratorType:      &shardIteratorType,
 		StartingSequenceNumber: ps,
 		StreamName:             aws.String(streamName),
+		Timestamp:              iteratorStartTimestamp,
+		// TODO: verify that providing this argument as explicitly nil doesn't affect other iterator types.
```
Comment (paulboocock) [inline shard_consumer.go:54]: Not setting it would mean it's `nil` so it shouldn't be any different if `iteratorStartTimestamp` is `nil`.

**Review by paulboocock — APPROVED** [review]
I haven't tested this but the code looks fine to me. Will test fully when its in a downstream app PR.

File: shard_consumer.go (hunk 1 above)
Comment (colmsnowplow) [reply shard_consumer.go:54]: Ah, I tested this and confirmed that you're right. Forgot to remove the comment. :)

**Comment by colmsnowplow** [comment]
> I haven't tested this but the code looks fine to me. Will test fully when its in a downstream app PR.

Here ya go: https://github.com/snowplow-devops/stream-replicator


I was just about to start fiddling with setting up a little integration test for this specific change in there while you reviewed. (and figure out how much effort it'd be to port something like that into this repo) :)

**Comment by colmsnowplow** [comment]
@paulboocock fyi I added an integration test for this feature to my test branch of the other repo: https://github.com/snowplow-devops/stream-replicator/tree/test-kinsumer-update

It's more complicated an instrumentation than I'd like, but if you're keen to give this a test you can do in that branch with:

```
make integration-up
cd pkg/source
go test -run TestKinesisSource_StartTimestamp
cd ../..
make integration-down
```

**Review by paulboocock — APPROVED** [review]

**Pull request merged**